### PR TITLE
Port to Coq 8.7

### PR DESCRIPTION
**Pull request by RalfJung**
Note that I have no idea what I am doing here. I patched the code (based on doc/changes.md) until it compiles...

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ clean: Makefile.coq
 	rm -f Makefile.coq
 
 Makefile.coq: _CoqProject
-	$(COQBIN)/coq_makefile -f _CoqProject -o Makefile.coq
+	coq_makefile -f _CoqProject -o Makefile.coq
```
Comment (ppedrot) [inline Makefile]: Please keep the variable in place, this allows portability of the script. You should rather export it in your environment.

Comment (RalfJung) [reply Makefile]: > You should rather export it in your environment.

Disagreed. Why do you want to insist on COQBIN being exported? I have coq in my PATH (managed by opam). Everything else works just fine, no need for COQBIN. Setting COQBIN will just lead to inconsistencies down the line.

Notice that further up in the Makefile, you already rely on coqc being in the PATH.

Comment (RalfJung) [reply Makefile]: In particular, I'd like to make this opam-ready and have it installable as a dependency. opam doesn't set COQBIN, so requiring that will not work.

Comment (RalfJung) [reply Makefile]: I found a snippet in ssreflect which makes it honor COQBIN if present, and fall back to the PATH otherwise.

**Pull request merged**